### PR TITLE
Added test_is_old_task_missing_healthchecks_with_real_data test

### PR DESCRIPTION
**Pull request by solarkennedy**
I was trying to make sure `is_old_task_missing_healthchecks` was working exactly like we thought it did, but it is a bit complex because it depends on the interface to marathon python.

From what I can tell it is still correct.

### CODE REVIEW

File: paasta_tools/autoscaling/autoscaling_service_lib.py [hunk 1]
```diff
@@ -802,11 +802,15 @@ def filter_autoscaling_tasks(
...
-                marathon_tasks[task.id] = task
+            if task.id.startswith(job_id_prefix):
+                if not app.health_checks:
+                    # If we don't have healthchecks defined anyway, then we should just
+                    # let the task through the filter
+                    marathon_tasks[task.id] = task
+                elif is_task_healthy(task) or is_old_task_missing_healthchecks(task, app):
+                    # healthy is good, but also we need to consider those old tasks missing
+                    # health checks, because those are still valid candidates for autoscaling
+                    marathon_tasks[task.id] = task
```
Comment (EvanKrall) [inline paasta_tools/autoscaling/autoscaling_service_lib.py]: I'm not sure I like the repetition of the task body here. What about:

```
if task.id.startswith(job_id_prefix):
    if any([
    ]):

File: tests/example-marathon-app-with-missing-healthchecks.json [hunk 2]
```diff
@@ -0,0 +1,450 @@
+{
```
Comment (EvanKrall) [inline tests/example-marathon-app-with-missing-healthchecks.json]: There's a lot of stuff in this json that isn't relevant to the test. Can you pare it down to only the parts necessary?

**Comment by EvanKrall** [comment]
If you think it works the same way as before, and this new test passes, what's the point of this change? Sounds like you haven't figured out what's wrong yet.

File: paasta_tools/autoscaling/autoscaling_service_lib.py (hunk 1 above)
Comment (solarkennedy) [reply paasta_tools/autoscaling/autoscaling_service_lib.py]: I agree that is better.
I couldn't understand the previous format, and I wasn't super happy with the nested ifs.

File: tests/example-marathon-app-with-missing-healthchecks.json (hunk 2 above)
Comment (solarkennedy) [reply tests/example-marathon-app-with-missing-healthchecks.json]: yes

**Comment by solarkennedy** [comment]
It is true, the point of this change is simply to increase test coverage and confidence that this function works.

**Pull request closed without merging**